### DISCUSSION TITLE
[v8] React Example: Editable Data gives error

### DISCUSSION THREAD

**Discussion by jonsveswe** [TanStack/table#4205] [General] [answered]
Hi. This example: [https://tanstack.com/table/v8/docs/examples/react/editable-data](https://tanstack.com/table/v8/docs/examples/react/editable-data) gives error:

"React Hook "React.useState" is called in function "cell" that is neither a React function component nor a custom React Hook function. React component names must start with an uppercase letter. (react-hooks/rules-of-hooks)"

"React Hook "React.useEffect" is called in function "cell" that is neither a React function component nor a custom React Hook function. React component names must start with an uppercase letter. (react-hooks/rules-of-hooks)"

It can be run in codesandbox but I can't run the example in Visual Studio Code. Any ideas how to fix this? 

(Edit: I can run it if I disable eslint but would like to find a solution without doing that.)

Best Regards

**Comment by tannerlinsley** [comment, marked as answer]
Any cell/header templates that use hooks can just use a classical function declaration:

```tsx
cell: function Cell () {
  const [state, setState ] = useState()
}
```

The function name should silence this issue.

**Reply by jonsveswe** [reply]
Ty!

**Reply by wsajjadh** [reply]
Worked, Thanks a lot.

**Reply by DanielSchranc** [reply]
Lovely! Thank you for your help. It would be nice to have this in documentation examples 🙏
